Neos.ContentGraph.DoctrineDbalAdapter/src/ContentRepositoryReadModelAdapter.php

```diff
@@ -56,11 +55,13 @@ public function findWorkspaceByName(WorkspaceName $workspaceName): ?Workspace
     {
         $workspaceByNameStatement = <<<SQL
             SELECT
-                name, baseWorkspaceName, currentContentStreamId, status
+                ws.name, ws.baseWorkspaceName, ws.currentContentStreamId, cs.sourceVersion as lastSourceVersion, scs.version as sourceVersion
             FROM
-                {$this->tableNames->workspace()}
+                {$this->tableNames->workspace()} ws
+                JOIN {$this->tableNames->contentStream()} cs ON cs.id = ws.currentcontentstreamid
+                LEFT JOIN {$this->tableNames->contentStream()} scs ON cs.sourceContentStreamId = scs.id
```

Comment:
so the performance trait is this: we still have the status in the workspace, but we have to calculate it slightly more expensive with these two joins.
I cannot say if this is better performance than having on the write side a similar complex update logic for each event. But as `findWorkspaceByName` (and implicitly  `getContentGraph`) trigger this query and are also part of the write side for constraint checks we have to carefully evaluate.

Reply:
At some point the content graph projections needs to know about changes between streams/workspaces (e.g. to count/diff pending changes etc). Then we can revisit this topic and make it faster if needed